venv_manager.plugin.zsh

```diff
@@ -70,10 +76,15 @@ function change_venv_if_needed() {
             deactivate_venv
         # switching venv environments
         elif [ $new_venv != $VIRTUAL_ENV ]; then
+            # force auto-deactivation here
+            _TMP_AUTO_DEACTIVATE_VENV=$_AUTO_DEACTIVATE_VENV
+            _AUTO_DEACTIVATE_VENV=1
             deactivate_venv
             activate_venv $new_venv after_deactivate
+            _AUTO_DEACTIVATE_VENV=$_TMP_AUTO_DEACTIVATE_VENV
```

Comment:
Is there a case where deactivating the previous env before activating the next is problematic?

Reply:
You always want to deactivate a venv before activating a new one right? This forces the auto-deactivation of a venv before the activation of a new one even if auto_deactivate is disabled.